hack/jenkins/dockerized-e2e-runner.sh

```diff
@@ -35,7 +35,6 @@ env \
   -u PATH \
   -u PWD \
   -u WORKSPACE \
-  -u GOROOT \
```

Comment:
I think adding this was correct, but reverting to the last known-good state seems safest.